### PR TITLE
Allow to configure language in cppcheck with a global var

### PR DESCRIPTION
**Pull request by ahcorde**
Allow to configure language in cppcheck with a global var

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake [hunk 1]
```diff
@@ -33,6 +33,12 @@ if(_source_files)
     list(APPEND _all_include_dirs ${ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS})
   endif()
 
+  # Language
+  set(_language "")
+  if(DEFINED ament_cmake_cppcheck_ADDITIONAL_LANGUAGE)
+    list(APPEND _language ${ament_cmake_cppcheck_ADDITIONAL_LANGUAGE})
```
Comment (dirk-thomas) [inline ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:39]: Doesn't the CMake function `ament_cppcheck` only accept a single language for `LANGUAGE`?

Comment (ahcorde) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:39]: cppcheck allows `c` or `c++` https://github.com/ament/ament_lint/blob/master/ament_cppcheck/ament_cppcheck/main.py#L68-L71

Comment (dirk-thomas) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:39]: > Doesn't the CMake function `ament_cppcheck` only accept **a single** language for `LANGUAGE`?

Comment (ahcorde) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake]: Fixed https://github.com/ament/ament_lint/pull/239/commits/d751f081f63d6977526fbd07b863a92f5d220593

Comment (dirk-thomas) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake]: The term "ADDITIONAL_LANGUAGE" doesn't make sense if only a single language can be selected.

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake [hunk 2]
```diff
@@ -33,6 +33,12 @@ if(_source_files)
     list(APPEND _all_include_dirs ${ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS})
   endif()
 
+  # Language
+  set(_language "")
+  if(DEFINED ament_cmake_cppcheck_LANGUAGE)
+    set(_language ${ament_cmake_cppcheck_LANGUAGE})
+  endif()
```
Comment (dirk-thomas) [inline ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:40]: This passes an empty `LANGUAGE` in the `else` case. I would argue it shouldn't pass `LANGUAGE` at all in that case.

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake [hunk 3]
```diff
@@ -33,6 +33,12 @@ if(_source_files)
     list(APPEND _all_include_dirs ${ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS})
   endif()
 
+  # Language
+  set(_language "c")
```
Comment (dirk-thomas) [inline ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:37]: Before this PR there is no default language and the tool determines the language automatically. That behavior should be preserved. If no language is specified no `LANGUAGE` / `--language` should be passed.

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake (hunk 2 above)
Comment (dirk-thomas) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake]: > and also not print a message in that case

This part of my comment is still pending

Comment (ahcorde) [reply ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake]: https://github.com/ament/ament_lint/pull/239/commits/67861398c1f4be21d0429ee3865a3e67633c17f9

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake [hunk 4]
```diff
@@ -33,7 +33,14 @@ if(_source_files)
     list(APPEND _all_include_dirs ${ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS})
   endif()
 
-  # Get exclde paths for added targets
+  # Language
+  set(_language "")
+  if(DEFINED ament_cmake_cppcheck_LANGUAGE)
+    set(_language LANGUAGE ${ament_cmake_cppcheck_LANGUAGE})
+    message(STATUS "Configured cppcheck language : ${_language}")
```
Comment (dirk-thomas) [inline ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:40]: The message should only print `ament_cmake_cppcheck_LANGUAGE` - not `_language`.

Nitpick: remove the space before the colon.

File: ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake [hunk 5]
```diff
@@ -33,7 +33,14 @@ if(_source_files)
     list(APPEND _all_include_dirs ${ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS})
   endif()
 
-  # Get exclde paths for added targets
+  # Language
```
Comment (dirk-thomas) [inline ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake:36]: Nitpick: not a valuable comment.

**Comment by ahcorde** [comment]
@dirk-thomas What is the process to merge this PR? I checked that this is running lint_cmake and copyright. There is no package upstream that uses this option yet (I plan to use this option in tf2). Should I run CI up to ament_lint and run these two tests?

**Comment by dirk-thomas** [comment]
CI builds testing the touched package seems to be enough in this case (since the patch shouldn't have any effect on existing packages).

**Comment by ahcorde** [comment]
Running CI up-to `ament_cmake_cppcheck` and `rcl`. Checking linters in `ament_cmake_cppcheck` and running the linter in one package.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10877/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6238/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8859/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10767/)

**Pull request merged**